### ISSUE TITLE
merge BAM files from multiple wf-basecalling instances

### ISSUE THREAD

**Issue by istvankleijn** [epi2me-labs/wf-somatic-variation#36]
### Is your feature related to a problem?

I attempted to run the workflow with the normal data coming from two separate (barcoded) sequencing runs. The two runs were basecalled (and demuxed) in two separate instances of wf-basecalling with the same settings. I created a folder with symlinks to the two BAM files and passed this folder to `--bam_normal`.

However, the workflow halted, saying it `Found mixed headers in (u)BAM files of sample [...]`. I checked the `@SQ` headers manually, and indeed these contain different links to the reference file. For example, the `chr1` lines are
```
@SQ     SN:chr1 LN:248956422    M5:6aef897c3d6ff0c78aff06ac189178dd     UR:/data/epi2me/instances/wf-basecalling_01J35KFCEPASK87ZKKNJ004QYW/work/09/65d11f352c54600388bcbbfc3d2d20/GRCh38-no_alt.fasta
```
and
```
@SQ     SN:chr1 LN:248956422    M5:6aef897c3d6ff0c78aff06ac189178dd     UR:/data/epi2me/instances/wf-basecalling_01J3QX5F0Y5K71SM46PBX7N81J/work/f3/e4350e08bf83d7d5ac9b92b98e1bfe/GRCh38-no_alt.fasta
```

### Describe the solution you'd like

Could the BAM header checks be made less strict so this use case is covered too?

### Describe alternatives you've considered

I now process the BAM files using `samtools` myself but this partially defeats the purpose of using the workflows with their containerization and controlled versioning. I would rather directly use the output from wf-basecalling as input to this workflow.

**Comment by SamStudio8** [comment]
@istvankleijn Thanks for raising this issue, it's another CRAM pitfall! It looks like our handling of SQ does not consider the UR tag when determining if headers are aligned to different references. I think it would it would be appropriate to just look at the SN/LN (and M5 if present) and ensure that a use-case like yours can still be permitted. I'll ticket this internally, I can't suggest when an update will be available at this time but we'll keep this ticket open.

**Comment by RenzoTale88** [comment]
@istvankleijn this should now be fixed in v1.4.0. Please give it a go and let us know if you still come across the same issue.

Andrea

**Comment by istvankleijn** [comment]
Yes it worked, thank you!
